homeassistant/components/switchbot_cloud/const.py

```diff
@@ -6,3 +6,4 @@
 DOMAIN: Final = "switchbot_cloud"
 ENTRY_TITLE = "SwitchBot Cloud"
 SCAN_INTERVAL = timedelta(seconds=600)
+LOCK_SCAN_INTERVAL = timedelta(seconds=30)
```

Comment:
The reason the scan interval is only once every 10 minutes is because of the limitation of the Switchbot API to 10000 calls a day: https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#request-limit

Calling the API every 600 seconds means that we can have up to 10000 / 24 / 6 = 69 devices (only for polling, we also want to actually control the devices so the actual number of devices we can handle is reduced, devices not requiring a feedback like IR devices are fine).

Calling the API every 30 seconds means that we can only have up to 3 devices.
With only 1 lock, it means that just for polling, we are using 2880 calls, for 2 locks, it is 5760 calls.

Supposing that a house only have 1 lock, using a lock with the current settings leaves 7120 calls, meaning up to 49 devices with the current scan interval. We are supposing that we are not controlling any device.

I believe the best solution is to add the ability to receive pushes from Switchbot.

Reply:
I fully agree with the reasoning regarding the poll intervals. If there is any way to hook into the Switchbot webhook system from a Home Assistant integration, I'd be happy to implement it. From my understanding of the API, it isn't easy to facilitate, since it would require the Home Assistant server to be reachable from the wider internet (and have a somewhat IP address/DynDNS setup). If there is some native integration available for this, please let me know and I'll take another look.

As an alternative/immediate solution, I would suggest setting the default intervals to 600 seconds for both locks and other devices but expose this setting to the user. I'm indifferent to whether it should be configurable per device or per device category, but I think it does make sense to have some things like locks be updated on a higher frequency.

Reply:
I am sorry, I do not have the knowledge about this, let us ask @joostlek as he often reviews this integration if he has an idea,

Reply:
@miterion Seems like [laurence-presland](https://github.com/laurence-presland) found a nice way to handle it  in https://github.com/home-assistant/core/pull/115522

Reply:
> As an alternative/immediate solution, I would suggest setting the default intervals to 600 seconds for both locks and other devices but expose this setting to the user.

We don't allow exposing/modifying the scan interval. They should be set to sane default values.
If the user wants a short update interval, they can use the service [`homeassistant.update_entity`](https://www.home-assistant.io/integrations/homeassistant/#service-homeassistantupdate_entity)